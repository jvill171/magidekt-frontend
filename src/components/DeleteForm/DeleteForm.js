
import { useState } from "react"
import "./DeleteForm.scss"


const DeleteForm = ({toDelete = "???", deleteAction}) =>{


    const [verifyDelTxt, setVerifyDelTxt ] = useState("")
    const toMatch = "Magidekt!"

    const handleDelInput = (e) => {
        setVerifyDelTxt(e.target.value);
    };

    const handleDelete = async (e)=>{
        e.preventDefault()
        await deleteAction();
    }

    return(
      <form className="DeleteForm">
        <h3>To delete your {toDelete}, type the following:</h3>
        <p>Magidekt!</p>
        <input type="text" onChange={handleDelInput} value={verifyDelTxt}/>
        <br/>
        <button
            className="delete-btn"
            onClick={handleDelete}
            disabled={toMatch !== verifyDelTxt}
        > Delete {toDelete} </button>
      </form>
    )
}

export default DeleteForm;