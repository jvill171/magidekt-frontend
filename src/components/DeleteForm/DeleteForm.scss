.DeckDelete{
    margin: 30px 0;
}