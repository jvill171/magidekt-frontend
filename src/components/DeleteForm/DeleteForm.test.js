import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import DeleteForm from './DeleteForm';

// Mock for deleteAction
const mockDeleteAction = jest.fn();

//Smoke test
test('renders DeleteForm component without crashing', () => {
    render(<DeleteForm
            toDelete="Test Item"
            deleteAction={mockDeleteAction} />);
});

//Snapshot test
test('DeleteForm component matches snapshot', () => {
    const { asFragment } = render(
    <DeleteForm
        toDelete="Test Item"
        deleteAction={mockDeleteAction} />
    );

    expect(asFragment()).toMatchSnapshot();
});
