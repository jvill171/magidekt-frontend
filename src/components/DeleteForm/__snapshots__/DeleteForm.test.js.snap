// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteForm component matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="DeleteForm"
  >
    <h3>
      To delete your Test Item, type the following:
    </h3>
    <p>
      Magidekt!
    </p>
    <input
      type="text"
      value=""
    />
    <br />
    <button
      class="delete-btn"
      disabled=""
    >
       Delete Test Item 
    </button>
  </form>
</DocumentFragment>
`;
